src/NLog/SetupSerializationBuilderExtensions.cs

```diff
@@ -0,0 +1,149 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog
+{
+    using System;
+    using System.Reflection;
+    using NLog.Common;
+    using NLog.Config;
+
+    /// <summary>
+    /// Extension methods to setup NLog extensions, so they are known when loading NLog LoggingConfiguration
+    /// </summary>
+    public static class SetupSerializationBuilderExtensions
+    {
+        /// <summary>
+        /// Overrides the active <see cref="IJsonConverter"/> with a new custom implementation
+        /// </summary>
+        public static ISetupSerializationBuilder RegisterJsonConverter(this ISetupSerializationBuilder setupBuilder, IJsonConverter jsonConverter)
+        {
+            ConfigurationItemFactory.Default.JsonConverter = jsonConverter ?? NLog.Targets.DefaultJsonSerializer.Instance;
+            return setupBuilder;
+        }
+
+        /// <summary>
+        /// Overrides the active <see cref="IValueFormatter"/> with a new custom implementation
+        /// </summary>
+        public static ISetupSerializationBuilder RegisterValueFormatter(this ISetupSerializationBuilder setupBuilder, IValueFormatter valueFormatter)
+        {
+            ConfigurationItemFactory.Default.ValueFormatter = valueFormatter ?? NLog.MessageTemplates.ValueFormatter.Instance;
+            return setupBuilder;
+        }
+
+        /// <summary>
+        /// Registers object Type transformation from dangerous (massive) object to safe (reduced) object
+        /// </summary>
+        public static ISetupSerializationBuilder RegisterObjectTransformation<T>(this ISetupSerializationBuilder setupBuilder, Func<T, object> transformer)
+        {
+            var original = ConfigurationItemFactory.Default.ObjectTypeTransformer;
+            ConfigurationItemFactory.Default.ObjectTypeTransformer = new ObjectTypeTransformation<T>(transformer, original);
+            return setupBuilder;
+        }
+
+        /// <summary>
+        /// Registers object Type transformation from dangerous (massive) object to safe (reduced) object
+        /// </summary>
+        public static ISetupSerializationBuilder RegisterObjectTransformation(this ISetupSerializationBuilder setupBuilder, string objectTypeName, Func<object, object> transformer)
```

Comment:
objectTypeName is the full qualified type name, isn't? Would `type` not a better option?

Reply:
Well it was actually done on requests, where the user wants to force tostring on certain objects without the project has to depend directly on the nuget-packages where object-types exists. See also https://github.com/NLog/NLog/pull/3610#issuecomment-537760596

Reply:
Clear. But think this is a bad idea. Of course he could fix this by passing the builder to s project which has a reference to it.

Reply:
If having the actual type, then one can use `RegisterObjectTransformation<T>`. I can add a bonus method that takes a Type as parameter?

Reply:
Btw. you are talking with the guy that is not going to use this feature at all. So not sure I will put up the best fight :) But it sounded really important for @mmurrell that requested this entire feature the first place. Kind of funny to only give half of the feature.

Reply:
Yeah sorry about that @mmurrell, but I think this weakens the API and it not strictly necessary.

Reply:
@304NotModified I guessing that the use of `string objectTypeName` will also be handy, when one day supporting configuration from NLog.config. Btw. it is not super-duper fully qualified, as it is without assembly name.

Reply:
@304NotModified Ignore my comment about it not being super-duper fully qualified. The `Type.GetType` is just lazy evaluated and it just disables the transformation if the type-name is wrong.

Reply:
Change from `string objectTypeName` to `Type objectType`